org.lflang/src/org/lflang/validation/ValidatorErrorReporter.java

```diff
@@ -0,0 +1,158 @@
+                ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return message;
+    }
+
+    /**
+     * Report the given message as an error on the given object.
+     */
+    @Override
+    public String reportError(EObject object, String message) {
+        errorsOccurred = true;
+        acceptor.acceptError(message, object, null,
+                ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return message;
+    }
+
+    /**
+     * Report the given message as an error on the current object.
+     * 
+     * Unfortunately, there is no way to provide a path and a line number to the
+     * ValidationMessageAcceptor as messages can only be reported directly as
+     * EObjects. While it is not an ideal solution, this method composes a
+     * messages indicating the location of the error and reports this on the
+     * object currently under validation. This way, the error message is not
+     * lost, but it is not necessarily reported precisely at the location of the
+     * actual error.
+     */
+    @Override
+    public String reportError(Path file, Integer line, String message) {
+        errorsOccurred = true;
+        String fullMessage = "Error in " + file.toString() + " on line "
+                + line.toString() + "\n" + message;
+        acceptor.acceptError(fullMessage, validatorState.getCurrentObject(),
+                null, ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return fullMessage;
+    }
+
+    /**
+     * Report the given message as a waring on the object currently under
+     * validation.
+     */
+    @Override
+    public String reportWarning(String message) {
+        acceptor.acceptWarning(message, validatorState.getCurrentObject(), null,
+                ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return message;
+    }
+
+    /**
+     * Report the given message as a warning on the given object.
+     */
+    @Override
+    public String reportWarning(EObject object, String message) {
+        acceptor.acceptWarning(message, object, null,
+                ValidationMessageAcceptor.INSIGNIFICANT_INDEX, null);
+        return message;
+    }
+
+
+    /**
+     * Report the given message as an warning on the current object.
+     * 
+     * Unfortunately, there is no way to provide a path and a line number to the
+     * ValidationMessageAcceptor as messages can only be reported directly as
+     * EObjects. While it is not an ideal solution, this method composes a
+     * messages indicating the location of the warning and reports this on the
+     * object currently under validation. This way, the warning message is not
+     * lost, but it is not necessarily reported precisely at the location of the
+     * actual warning.
+     */
+    @Override
+    public String reportWarning(Path file, Integer line, String message) {
+        String fullMessage = "Warning in " + file.toString() + " on line "
```

Comment:
I think the convention has been `WARNING: ...` for warnings.

Reply:
Same as above.